nano/node/common.cpp

```diff
@@ -500,14 +547,63 @@ void nano::confirm_req::visit (nano::message_visitor & visitor_a) const
 
 void nano::confirm_req::serialize (nano::stream & stream_a) const
 {
-	assert (block != nullptr);
 	header.serialize (stream_a);
-	block->serialize (stream_a);
+	if (header.block_type () == nano::block_type::not_a_block)
+	{
+		assert (!roots_hashes.empty ());
+		// Calculate size
+		uint8_t count (roots_hashes.size ());
+		write (stream_a, count);
+		assert (count <= 32);
```

Comment:
If roots_hashes.size () is 257 then count will be 1 and this assert won't be firing correctly. Should this instead be on roots_hashes.size ()?

Reply:
Good point